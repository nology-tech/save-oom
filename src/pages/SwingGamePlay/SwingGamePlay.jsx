import React from "react";
// import background from '../../assets/images/GameBackground.png';
import swingingOom from "../../assets/images/Group 146swingingOom.png";
import squirrel from "../../assets/images/squirrel.png";
import AnimatedImage from "../../components/AnimatedImage/AnimatedImage";
import "./SwingGamePlay.scss";

const SwingGamePlay = () => {
  return (
    <div className="swing-game-play">
      {/* <img
        className='swing-game-play__background'
        src={background}
        alt='background'
      /> */}

      <AnimatedImage
        imageToAnimate={squirrel}
        animationClass={"animate__animated"}
        animationType={"animate__bounce"}
        imageStylesClass={"swing-game-play__squirrel"}
      />
      <AnimatedImage
        imageToAnimate={swingingOom}
        animationClass={"animate__animated.animate__fast"}
        animationType={"animate__swing"}
        imageStylesClass={"swing-game-play__oom"}
      />
    </div>
  );
};

export default SwingGamePlay;
