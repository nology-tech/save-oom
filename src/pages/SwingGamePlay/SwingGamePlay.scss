@use '../../styles/variables' as *;
@use '../../../node_modules/animate.css/animate.css';

.swing-game-play {
  // display: flex;
  // flex-direction: row;
  width: 100%;
  background-image: url(../../assets/images/GameBackground.png);
  background-size: 100% 100%;
  background-repeat: no-repeat;

  // &__background {
  //   background-repeat: no-repeat;
  //   background-attachment: fixed;
  //   background-size: cover;
  //   height: 100%;
  //   width: 100%;
  // }

  &__phonic {
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
  }

  &__button-and-score {
    // border: 2px solid red;
    margin-top: 7rem;
    justify-self: center;
    display: flex;
    text-align: center;
    align-items: flex-end;
    padding: 2rem;

    p {
      color: $color-white;
    }
    
  }

  &__oom {
    position: absolute;
    // top: 32.5rem;
    // left: 65rem;
    bottom: 0rem;
    left: 63%;
    animation-delay: 1s;
    animation-duration: 8s;
  }

  &__squirrel {
    position: absolute;
    // left: 20rem;
    // top: 37rem;
    bottom: 1rem;
    left: 30%;
    animation-duration: 10s;
    animation-delay: 1s;
  }
}
