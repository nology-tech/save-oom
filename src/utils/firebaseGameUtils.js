import { db } from "../firebase";
import {
  addDoc,
  collection,
  getDocs,
  query,
  where,
  orderBy,
  Timestamp,
} from "firebase/firestore";
/** 
 * Get all the rounds of a specific game and level for a given user, ordered by most recent first 
 @param {*} userId
  @param {String} game
*@returns a querySnap (document snapshot), returns a Promise.
*/

export const getGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** 
   * Get all the correct round of a specific game and level for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns a promise with the query snapshot.
  */
export const getCorrectGameRoundsForUser = async (id, game, level) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("level", "==", level),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect rounds for a given user, ordered by most recent first 
  @param {String} userId
  *@returns  a promise with the query snapshot.
  */
export const getIncorrectGameRoundsForUser = async (id) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("correct", "==", false),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};

/** Get all the incorrect round of a specific game for a given user, ordered by most recent first 
  @param {String} userId
  @param {String} gameId
  @param {Number} level  
  *@returns  a promise with the query snapshot.
  */

export const getAllIncorrectGameRoundsForUser = async (id, game) => {
  const q = query(
    collection(db, "users", id, "rounds_played"),
    where("gameId", "==", game),
    where("correct", "==", true),
    orderBy("roundTime", "desc")
  );
  const querySnap = await getDocs(q);
  return querySnap;
};



/** Function that returns an array of rounds 
  @param {String} userId
  @param {String} gameId
  @param {Function} getGameRounds
  *@returns returs a promise.
  */
export const getArrayOfRounds = async (id, game, level, getGameRounds) => {
  const querySnap = await getGameRounds(id, game, level);
  const phonicsArr = querySnap.docs.map((doc) => {
    return doc.data().phonic;
  });
  return phonicsArr;
};

/** Function that returns an array of rounds 
  @param {String} id
  @param {String} game
  @param {Object} roundInfo
  @param {String} answer
  *@returns returs a promise.
  */

export const saveUserRound = async (id, game, level, gameStats, answer) => {
  const { isCorrect, score } = gameStats;
  await addDoc(collection(db, "users", id, "rounds_played"), {
    correct: isCorrect,
    gameId: game,
    phonic: answer,
    level: level,
    roundTime: Timestamp.fromDate(new Date()).toDate(),
    score: score,
  });
};
