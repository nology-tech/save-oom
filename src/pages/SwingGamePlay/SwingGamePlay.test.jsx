import { customRender } from '../../utils/testUtils';
import SwingGamePlay from './SwingGamePlay';
import { render, screen } from '@testing-library/react';

it("Should render the Swing Game Play screen", () => {
  const { container } = customRender(<SwingGamePlay />);

  expect(container).toMatchSnapshot();
});

it('Should render the score count screen', () => {
  render(<SwingGamePlay />);

  const scoreCount = screen.getByText(/Number Of Correct Sounds:/i);

  expect(scoreCount).toBeInTheDocument();
});
