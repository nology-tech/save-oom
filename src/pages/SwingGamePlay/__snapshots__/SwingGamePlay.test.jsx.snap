// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the Swing Game Play screen 1`] = `
<div>
  <div
    class="swing-game-play"
  >
    <div
      class="container"
    >
      <div
        class="needs"
      >
        <div
          class="needs__image-div"
        >
          <img
            alt="ooms needs"
            class="needs__image"
            src="Battery-0.svg"
          />
        </div>
      </div>
      <p
        class="needs__percentage"
      >
        0
        %
      </p>
      <div
        class="needs"
      >
        <div
          class="needs__image-div"
        >
          <img
            alt="ooms needs"
            class="needs__image"
            src="Smiley.svg"
          />
        </div>
      </div>
      <p
        class="needs__percentage"
      >
        0
        %
      </p>
      <div
        class="needs"
      >
        <div
          class="needs__image-div"
        >
          <img
            alt="ooms needs"
            class="needs__image"
            src="Battery-20.svg"
          />
        </div>
      </div>
      <p
        class="needs__percentage"
      >
        0
        %
      </p>
    </div>
    <div
      class="swing-game-play__phonic"
    >
      <h1
        class="timer"
      >
        60
        <span
          class="timer__s"
        >
          s
        </span>
      </h1>
      <div
        class="phonic-component"
      >
        <h1
          class="phonic-component__heading"
        >
          s
        </h1>
      </div>
    </div>
    <div>
      <img
        alt="Animated Image"
        class="animate__animated swing-game-play__squirrel  "
        src="squirrel.png"
      />
    </div>
    <img
      alt="Animated Image"
      class="animate__animated.animate__fastest swing-game-play__oom "
      src="Group 146swingingOom.png"
    />
    <div
      class="validation"
    >
      <button
        class="validation__button validation__button--wrong"
      >
        <img
          alt="cross"
          src="Cross.png"
        />
      </button>
      <button
        class="validation__button validation__button--right"
      >
        <img
          alt="tick"
          src="Tick.png"
        />
      </button>
    </div>
    <p
      class="swing-game-play__score"
    >
      Number Of Correct Sounds: 
      0
    </p>
  </div>
</div>
`;
